### PR TITLE
[MAP-101] Add Golang request signature example

### PR DESCRIPTION
**Pull request by tl-marco-tormento**
#51 for Golang

# Golang request signature example
Sends a signed request to `https://api.truelayer-sandbox.com/test-signature`.

## Run
Set environment variables:
* `ACCESS_TOKEN` A valid JWT access token for `payments` scope [docs](https://docs.truelayer.com/docs/retrieve-a-token-in-your-server).
* `KID` The certificate/key UUID for associated with your public key uploaded to console.truelayer.com.
* `PRIVATE_KEY` Private key PEM string that matches the `KID` & uploaded public key.
  Should have the same format as [this example private key](https://github.com/TrueLayer/truelayer-signing/blob/main/test-resources/ec512-private.pem).

```sh
$ go run main.go
Sending...
204 ✓
```

### CODE REVIEW

**Review by alexheretic — APPROVED** [review]
lgtm

```
$ go run main.go
Sending...
204 ✓
```

**Review by azanin — APPROVED** [review]

**Review by conradludgate — APPROVED** [review]

File: go/examples/sign-request/main.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			responseBody = []byte("✓")
+		} else {
+			defer resp.Body.Close()
+			responseBody, err = io.ReadAll(resp.Body)
+			if err != nil {
+				responseBody = []byte(fmt.Sprintf("%s", err.Error()))
+			}
+		}
+	} else {
+		responseBody = []byte(fmt.Sprintf("%s", err.Error()))
```
Comment (conradludgate) [inline go/examples/sign-request/main.go:80]: Why are you formatting these?

File: go/examples/sign-request/main.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	"os"
+	"strings"
+
+	tlsigning "github.com/Truelayer/truelayer-signing/go"
+	"github.com/google/uuid"
+)
+
+// the base url to use
+const (
+	TlBaseUrl = "https://api.truelayer-sandbox.com"
```
Comment (conradludgate) [inline go/examples/sign-request/main.go:17]: Go usually uses full-upper acronyms if I remember correctly
```suggestion
	TLBaseURL = "https://api.truelayer-sandbox.com"
```

File: go/examples/sign-request/main.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		Header("Idempotency-Key", []byte(idempotencyKey)).
+		Header("X-Bar-Header", []byte("abc123")).
+		Body([]byte(body)). // body of our request
+		Sign()
+
+	fmt.Println("Sending...")
+
+	// Request body & any signed headers *must* exactly match what was used to generate the signature.
+	client := &http.Client{}
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/test-signature", TlBaseUrl), strings.NewReader(body))
```
Comment (conradludgate) [inline go/examples/sign-request/main.go:59]: Unchecked error ;)

File: go/examples/sign-request/main.go (hunk 1 above)
Comment (skos-ninja) [reply go/examples/sign-request/main.go:80]: Yeah in this case `.Error()` is a string so shouldn't need to be formatted. There's also should we be converting the strings to bytes and then back to strings again or just converting the response body to string and then using that below

File: go/examples/sign-request/main.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	// A random body string is enough for this request as `/test-signature` endpoint does not
+	// require any schema, it simply checks the signature is valid against what's received.
+	body := fmt.Sprintf("body-%d", rand.Intn(99999999))
+
+	idempotencyKey := uuid.New().String()
+
+	// Generate tl-signature
+	tlSignature, err := tlsigning.SignWithPem(kid, []byte(privateKey)).
```
Comment (skos-ninja) [inline go/examples/sign-request/main.go:45]: Unchecked error here too

Comment (tl-marco-tormento) [reply go/examples/sign-request/main.go:45]: Our signing library never fails. 😎

**Pull request merged**